### PR TITLE
Add methods for getting default attributes for item

### LINKED ISSUE #5587: ItemMeta doesn't include vanilla attributes added in nms overrides
### Expected behavior

Calling ItemMeta.getAttributeModifiers() or any of its overloads returns all AttributeModifiers on an item.

### Observed/Actual behavior

If a vanilla item adds AttributeModifiers by overriding vanilla's getAttributeModifiers method (`Item.a(EnumItemSlot)`), they are not properly wrapped and returned by ItemMeta, as ItemMeta is only checking the item's NBT for AttributeModifiers.

### Steps/models to reproduce

Call `ItemStack.getItemMeta().getAttributeModifiers()` or any of its overloads on an item that, in vanilla, has AttributeModifiers built into the item (ex. tools, armor, swords, trident).

Using the test plugin, log into the server and hold a (tools, armor, swords, trident) and run the `/test` command. `null` will be printed in the console, despite the player being able to mouse over the item and see the tooltip from attackdamage/attackspeed/armor AttributeModifiers.

### Plugin list

[TestPlugin.zip](https://github.com/PaperMC/Paper/files/6438204/TestPlugin.zip)
```
public class TestPlugin extends JavaPlugin implements CommandExecutor {
    @Override
    public boolean onCommand(CommandSender cs, Command cmnd, String string, String[] args) {
        Player p = (Player) cs;
        ItemStack heldItem = p.getInventory().getItemInMainHand();
        ItemMeta meta = heldItem.getItemMeta();
        if (meta != null) {
            System.out.println(meta.getAttributeModifiers());
        }
        return true;
    }   
}
```

### Paper version

> version
[19:55:16 INFO]: Checking version, please wait...
[19:55:17 INFO]: This server is running Paper version git-Paper-647 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)
You are 1 version(s) behind
Previous version: git-Paper-631 (MC: 1.16.5)

### Agreements

- [x] I am running the latest version of Paper available from https://papermc.io/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] My version of Minecraft is supported by Paper.

### Other

/version reports I'm 1 version behind, latest available on papermc.io is what I'm running.

Referencing #5311 here, I'm trying to make a plugin that makes use of vanilla damage attributes on an item (ie I need to know how much base damage it will do when hitting an entity). There is not currently any API method for calculating an Attribute's final value, so I am forced to check the mc wiki and do the calculation myself. However, since the API currently does not list AttributeModifiers added via nms method override, I am forced to use nms/reflection to get that information.

### PR DESCRIPTION
**Pull request by Machine-Maker**
Closes #5587 (I think)
touches #5311

### CODE REVIEW

**Comment by kennytv** [comment]
Why in unsafe?

**Comment by Machine-Maker** [comment]
Well, one not-so-good reason is, that there isn't another good place for it? You need to be able to get the attributes from just a normal ItemStack, not a CraftItemStack, and ItemStack doesn't have access to the nms Item.

EDIT: It could also go on the Bukkit interface (the methods for getting attributes since that has a similar functionality to Unsafe in this respect, but see below reason for why I put it on Unsafe).

The other is, that I put the rarity stuff in Unsafe and that was merged, so I thought this was along the same lines.

EDIT: Also, the translation keys for materials are also there. Basically, since it only applies to some of the Material enum values, I thought Unsafe would be the best place. Ideally, once that's nuked, none of those are "unsafe".

File: Spigot-Server-Patches/0723-Attributes-API-for-item-defaults.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+     }
+ 
++    @Override
++    public com.google.common.collect.Multimap<org.bukkit.attribute.Attribute, org.bukkit.attribute.AttributeModifier> getItemAttributes(org.bukkit.Material material, org.bukkit.inventory.EquipmentSlot equipmentSlot) {
++        Item item = getItem(material);
++        if (item == null) {
++            throw new IllegalArgumentException(material + " is not an item and therefore does not have attributes");
++        }
++        com.google.common.collect.ImmutableMultimap.Builder<org.bukkit.attribute.Attribute, org.bukkit.attribute.AttributeModifier> attributeMapBuilder = com.google.common.collect.ImmutableMultimap.builder();
++        item.a(org.bukkit.craftbukkit.CraftEquipmentSlot.getNMS(equipmentSlot)).forEach((attributeBase, attributeModifier) -> {
```
Comment (kashike) [inline Spigot-Server-Patches/0723-Attributes-API-for-item-defaults.patch]: Use an obfuscation helper.

**Review by kashike — DISMISSED** [review]

File: Spigot-Server-Patches/0723-Attributes-API-for-item-defaults.patch [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+index 6141e86278d876e42dbed6e8f2275280babcef77..af136518a19c09a0c4bc81da0d28b906016b7137 100644
+--- a/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
++++ b/src/main/java/org/bukkit/craftbukkit/util/CraftMagicNumbers.java
+@@ -481,6 +481,19 @@ public final class CraftMagicNumbers implements UnsafeValues {
+         return io.papermc.paper.inventory.ItemRarity.values()[getItem(itemStack.getType()).getItemStackRarity(CraftItemStack.asNMSCopy(itemStack)).ordinal()];
+     }
+ 
++    @Override
++    public com.google.common.collect.Multimap<org.bukkit.attribute.Attribute, org.bukkit.attribute.AttributeModifier> getItemAttributes(org.bukkit.Material material, org.bukkit.inventory.EquipmentSlot equipmentSlot) {
++        Item item = getItem(material);
```
Comment (Proximyst) [inline Spigot-Server-Patches/0723-Attributes-API-for-item-defaults.patch]:
```suggestion
+        Item item = this.getItem(material);
```
Let's make Den have less work to do :)

**Review by Proximyst — APPROVED** [review]
LGTM.

Best in unsafe: any methods that require implementation but don't really have decent locations for them (i.e. interfaces or abstract classes, so ItemStack and Material as the only good locations are off the table) go there.

**Comment by Machine-Maker** [comment]
Implemented suggestions

**Review by kennytv — APPROVED** [review]
Thenk

**Pull request merged**